internal/impl/mysql/input_mysql_stream.go

```diff
@@ -0,0 +1,645 @@
+			if err != nil {
+				if !errors.Is(cErr, service.ErrKeyNotFound) {
+					res.Logger().Errorf("failed to obtain cursor cache item. %v", cErr)
+				}
+				return
+			}
+
+			var storedMySQLBinLogPosition mysqlReplications.Position
+			if err = json.Unmarshal(binLogPositionBytes, &storedMySQLBinLogPosition); err != nil {
+				res.Logger().With("error", err.Error()).Error("Failed to unmarshal stored binlog position.")
+				return
+			}
+
+			streamInput.startBinLogPosition = &storedMySQLBinLogPosition
+		}); err != nil {
+
+			res.Logger().With("error", err.Error()).Error("Failed to access cursor cache.")
+		}
+	}
+
+	i := &streamInput
+	i.cp = checkpoint.NewCapped[*int64](int64(i.checkPointLimit))
+
+	i.tablesFilterMap = map[string]bool{}
+	for _, table := range i.tables {
+		if err = validateTableName(table); err != nil {
+			return nil, err
+		}
+		i.tablesFilterMap[table] = true
+	}
+
+	res.Logger().Info("Starting MySQL stream input")
+
+	if batching, err = conf.FieldBatchPolicy(fieldBatching); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	i.batching = batching
+	if i.batchPolicy, err = i.batching.NewBatcher(res); err != nil {
+		return nil, err
+	} else if batching.IsNoop() {
+		batching.Count = 1
+	}
+
+	r, err := service.AutoRetryNacksBatchedToggled(conf, i)
+	if err != nil {
+		return nil, err
+	}
+
+	return conf.WrapBatchInputExtractTracingSpanMapping("mysql_stream", r)
+}
+
+func init() {
+	err := service.RegisterBatchInput(
+		"mysql_stream", mysqlStreamConfigSpec,
+		func(conf *service.ParsedConfig, mgr *service.Resources) (service.BatchInput, error) {
+			return newMySQLStreamInput(conf, mgr)
+		})
+	if err != nil {
+		panic(err)
+	}
+}
+
+// ---- Redpanda Connect specific methods----
+
+func (i *mysqlStreamInput) Connect(ctx context.Context) error {
+	canalConfig := canal.NewDefaultConfig()
+	canalConfig.Flavor = i.flavor
+	canalConfig.Addr = i.mysqlConfig.Addr
+	canalConfig.User = i.mysqlConfig.User
+	canalConfig.Password = i.mysqlConfig.Passwd
+	// resetting dump path since we are doing snapshot manually
+	// this is required since canal will try to prepare dumper on init stage
+	canalConfig.Dump.ExecutionPath = ""
+
+	// Parse and set additional parameters
+	canalConfig.Charset = i.mysqlConfig.Collation
+	if i.mysqlConfig.TLS != nil {
+		canalConfig.TLSConfig = i.mysqlConfig.TLS
+	}
+	canalConfig.ParseTime = true
+	canalConfig.IncludeTableRegex = i.tables
+
+	c, err := canal.NewCanal(canalConfig)
+	if err != nil {
+		return err
+	}
+	c.AddDumpTables(i.mysqlConfig.DBName, i.tables...)
+
+	i.canal = c
+
+	db, err := sql.Open("mysql", i.dsn)
+	if err != nil {
+		return err
+	}
+
+	// create snapshot instance
+	snapshot := NewSnapshot(ctx, i.logger, db)
+	i.snapshot = snapshot
+
+	go i.startMySQLSync(ctx)
+	return nil
+}
+
+func (i *mysqlStreamInput) readMessages(ctx context.Context) {
+	var nextTimedBatchChan <-chan time.Time
+	var latestPos *mysqlReplications.Position
```

Comment:
It's confusing to me that sometimes we use this variable and sometimes we use `i.currentLogPosition` to use for acking. Can we always use `i.currentLogPosition`?